converter/shieldhit/beam.py

```diff
@@ -1,19 +1,28 @@
 from dataclasses import dataclass
 import math as m
+from enum import Enum
+
+class DefinitionType(Enum):
+
+    SIMPLE = "simple"
+
+    FILE = "file"
 
 
 @dataclass
 class BeamConfig:
     """Class mapping of the beam.dat config file."""
 
-    energy: float = 150.
+    energy: float = 150.0
     energy_spread: float = 1.5
     beam_ext_x: float = -0.1
     beam_ext_y: float = 0.1
     delta_e: float = 0.03
     nstat: int = 10000
     beampos: tuple[float, float, float] = (0, 0, 0)
     beamdir: tuple[float, float, float] = (0, 0, 1)
+    definition_type: DefinitionType = DefinitionType.SIMPLE
+    definition_file: str = None
```

Comment:
```suggestion
    definition_file: Optional[str] = None
```

also add import for Optional from pyparsing